shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs

```diff
@@ -225,6 +244,54 @@ maybeToStrictMaybe :: Maybe a -> StrictMaybe a
 maybeToStrictMaybe Nothing  = SNothing
 maybeToStrictMaybe (Just x) = SJust x
 
+fromSMaybe :: a -> StrictMaybe a -> a
+fromSMaybe d SNothing = d
+fromSMaybe _ (SJust x) = x
+
+newtype Url = Url { unUrl :: Text64 }
+  deriving (Eq, Generic, Show, ToCBOR, FromCBOR, NoUnexpectedThunks)
+
+mkUrl :: Text -> Url
+mkUrl = Url . text64
+
+newtype DnsName = DnsName { unDnsName :: Text64 }
+  deriving (Eq, Generic, Show, ToCBOR, FromCBOR, NoUnexpectedThunks)
+
+mkDnsName :: Text -> DnsName
+mkDnsName = DnsName . text64
+
+dnsSize :: DnsName -> Integer
+dnsSize = toInteger . text64Size . unDnsName
+
+newtype IPv4 = IPv4 { unIPv4 :: ByteString }
+  deriving (Eq, Generic, Show, ToCBOR, NoUnexpectedThunks)
+
+mkIPv4 :: Word8 -> Word8 -> Word8 -> Word8 -> IPv4
```

Comment:
This interface is a bit cumbersome.
I propose that you use `HostAddress` and `HostAddress6` from the network package.
`mkIPv4 :: HostAddress -> IPv4`
`mkIPv6 :: HostAddress6 -> IPv6`.

Reply:
sounds good, thank you @karknu!